File: examples/python-examples/CollisionPrecise.dox.py
from meshlib import mrmeshpy as mm

meshA = mm.makeUVSphere() # make mesh A
meshB = mm.makeUVSphere() # make mesh B
meshB.transform(mm.AffineXf3f.translation(mm.Vector3f(0.1,0.1,0.1))) # shift mesh B for better demonstration

converters = mm.getVectorConverters(meshA,meshB) # create converters to integer field (needed for absolute precision predicates)
collidingFaceEdges = mm.findCollidingEdgeTrisPrecise(meshA,meshB,converters.toInt) # find each intersecting edge/triangle pair 
# print pairs of edges tirangles
for vet in collidingFaceEdges:
    if vet.isEdgeATriB():
        print("edgeA:",vet.edge.get(),"triB:",vet.tri().get())
    else:
        print("triA:",vet.tri().get(),"edgeB:",vet.edge.get())

File: examples/js-examples/CollisionPrecise.dox.js
import createMeshLib from '@meshinspector/meshlib';

const ml = await createMeshLib();

using meshA = ml.makeUVSphere(1.0, 16, 16); // make mesh A
using meshB = ml.makeUVSphere(1.0, 16, 16); // make mesh B
// shift mesh B for better demonstration
using shift = ml.AffineXf3f.translation({ x: 0.1, y: 0.1, z: 0.1 });
meshB.transform(shift);

// create converters to integer field (needed for absolute precision predicates)
using converters = ml.getVectorConverters(meshA, meshB);
// find each intersecting edge/triangle pair
using collidingFaceEdges = ml.findCollidingEdgeTrisPrecise(meshA, meshB, converters);
for (let i = 0; i < collidingFaceEdges.size(); ++i) {
  using vet = collidingFaceEdges.get(i);
  if (vet.isEdgeATriB())
    console.log(`edgeA: ${vet.edge}, triB: ${vet.tri()}`);
  else
    console.log(`triA: ${vet.tri()}, edgeB: ${vet.edge}`);
}
